server/webex_store.go

```diff
@@ -0,0 +1,270 @@
+	}
+
+	if session.Token.AccessToken == "" {
+		return ErrUnableToSaveSessionMissingAccessToken
+	}
+
+	config := p.getConfiguration()
+
+	accessToken, encryptErr := encrypt([]byte(config.EncryptionKey), session.Token.AccessToken)
+	if encryptErr != nil {
+		return encryptErr
+	}
+
+	session.Token.AccessToken = accessToken
+
+	// Encrypt the refresh token if it's present
+	if session.Token.RefreshToken != "" {
+
+		refreshToken, err := encrypt([]byte(config.EncryptionKey), session.Token.RefreshToken)
+		if err != nil {
+			return err
+		}
+
+		session.Token.RefreshToken = refreshToken
+	}
+
+	data, err := json.Marshal(session)
+	if err != nil {
+		return err
+	}
+
+	if err := p.API.KVSet(fmt.Sprintf("%s%s", WebexOAuthSessionKey, session.UserID), data); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+// loadWebexSession retrieves the webex session if present from the KV store
+func (p *Plugin) loadWebexSession(userID string) (WebexOAuthSession, error) {
+
+	session := WebexOAuthSession{}
+
+	key := fmt.Sprintf("%s%s", WebexOAuthSessionKey, userID)
+
+	data, appErr := p.API.KVGet(key)
+	if appErr != nil {
+		p.API.LogError(
+			"Error retrieving webex session",
+			"user_id", userID,
+			"key", key,
+			"error", appErr.Error(),
+		)
+		return session, appErr
+	}
+
+	if len(data) == 0 {
+		return session, ErrWebexSessionNotFound
+	}
+
+	err := json.Unmarshal(data, &session)
+	if err != nil {
+		return session, err
+	}
+
+	config := p.getConfiguration()
+
+	if time.Now().UTC().After(session.Token.Expiry) {
+		kvErr := p.API.KVDelete(key)
+		if kvErr != nil {
+			return WebexOAuthSession{}, kvErr
+		}
+		return WebexOAuthSession{}, ErrWebexSessionExpired
+	}
+
+	accessToken, err := decrypt([]byte(config.EncryptionKey), session.Token.AccessToken)
+	if err != nil {
+		return session, err
+	}
+
+	session.Token.AccessToken = accessToken
+
+	// Decrypt refresh token if it's present
+	if session.Token.RefreshToken != "" {
+
+		refreshToken, err := decrypt([]byte(config.EncryptionKey), session.Token.RefreshToken)
+		if err != nil {
+			return session, err
+		}
+
+		session.Token.RefreshToken = refreshToken
+	}
+
+	return session, nil
+}
+
+// getWebexUserInfo retrieves the webex user info from the KV store and
+// if not present will request from the webex API and subsequently save
+// it to the KV store if successful
+func (p *Plugin) getWebexUserInfo(userID string) (WebexUserInfo, error) {
+
+	webexUser, loadErr := p.loadWebexUser(userID)
+
+	switch {
+	case loadErr == ErrWebexUserNotFound:
+
+		session, err := p.loadWebexSession(userID)
+		if err != nil {
+			p.API.LogError("error looking up session", "error", err.Error())
+			return webexUser, err
+		}
+
+		webex, err := NewWebexClient(session.Token.AccessToken)
+		if err != nil {
+			p.API.LogError("Error creating new webex client", "error", err.Error())
+			return webexUser, err
+		}
+
+		// unlikely, but just in case to avoid angry panics
+		if webex == nil {
```

Comment:
The `NewWebexClient` always return a valid client, never a nil, so is safe to skip this check here.

Reply:
Removed